### DISCUSSION TITLE
Custom Civilian Units

### DISCUSSION THREAD

**Discussion by Tetlys** [Vdauphin/HeartsAndMinds#1299] [Question and answer] [answered]
Hi there, 

Is there any way I can select custom civilian units to spawn, I recognise that I can currently choose a faction, i'm looking to use a specific group of  Africans Civilians from Rotators collection which currently count as "civilian other" I belive. I see you have a case file for adding and removing units for _p_en, but not for _p_civ? How would I go about adding this? or is there somewhere/way I could do this easier already? 

Thanks!

**Comment by Vdauphin** [comment, marked as answer]
Hello,

> Is there any way I can select custom civilian units to spawn, I recognise that I can currently choose a faction, i'm looking to use a specific group of Africans Civilians from Rotators collection which currently count as "civilian other" I belive. 

You can select in mission parameter any faction used by optionals mods (https://github.com/Vdauphin/HeartsAndMinds/releases)

>  but not for _p_civ? 

Yea I don't have because most of the time you don't need to tweak civilians class names

> How would I go about adding this? 

Just replace by an array of all class names you want to use here https://github.com/Vdauphin/HeartsAndMinds/blob/3869216e1389c48a8830c18cfe79d03472b02587/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/mission.sqf#L364

Same for vehicle: https://github.com/Vdauphin/HeartsAndMinds/blob/3869216e1389c48a8830c18cfe79d03472b02587/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/mission.sqf#L366

Cheers
